XMonad/Actions/TreeSelect.hs

```diff
@@ -656,7 +656,7 @@ drawStringXMF display window visual colormap gc font col x y text = case font of
 -- Note that it uses short to represent its components
 fromARGB :: Pixel -> XRenderColor
 fromARGB x = XRenderColor (fromIntegral $ 0xff00 .&. shiftR x 8)  -- red
-                          (fromIntegral $ 0xff00 .&. x)           -- green
-                          (fromIntegral $ 0xff00 .&. shiftL x 8)  -- blue
+                          (fromIntegral $ 0xff00 .&. shiftL x 8)  -- green
+                          (fromIntegral $ 0xff00 .&. x)           -- blue
```

Comment:
Oh crap. :-(
This is actually a bug in https://hackage.haskell.org/package/X11-xft-0.3.1/src/Graphics/X11/Xrender.hsc, but that package is probably unmaintained, so we may need to work around it here, although it'd be nice to at least try contacting Clemens. :-/

Turns out the same problem is in xmobar: when alpha is less than 255 and bgcolor is green, the xft-drawn parts of xmobar are blue and the rest is green. It's really silly.

We definitely need a comment explaining why the colors are swapped here.

Reply:
As far as I tested this fixes the foreground issue and does not break anything, but I might try to fix the Xrender thing as well.

Reply:
Yeah, I tested this as well, because it all seemed very strange, and that led to me to the discovery of the root cause in X11-xft and the xmobar issue as well. If you decide to contact Clemens, feel free to Cc me. In any case, we need a comment here explaining what's going on.

Reply:
What comment should I add?

Reply:
One that summarizes what was said in this comment thread. If you don't know what to add, I guess I can do it later. :-/